pagopa/io-app
# chore: [IOPID-1787] Add `TRACKING`  Profile Property

## Short description
This PR adds the `TRACKING` profile property as described [here](https://pagopa.atlassian.net/wiki/spaces/CM1/pages/944242689/TPR+Integrazioni+Login+e+Onboarding+in+appIO).

## List of changes proposed in this pull request
- Added `TRACKING`  profile property.
- Added some utility functions to save the new profile property.
- Tried to fix some edge cases that happen during opt-in/opt-out events by switching function calls and add some timings.

## How to test
Run the application and change your permission about sharing or not MP data. Check on MP if the profile property is updated correctly and if all the events are as expected.

See [here](https://eu.mixpanel.com/project/2460815/view/3003771/app/profile#distinct_id=2A3F1734-A6CE-4878-B1F4-575E19346E3E) on 13/06/2024, to check the events flow collected during development.
